cpp/include/cudf/types.h

```diff
@@ -85,6 +112,24 @@ typedef struct {
 	// here we can also hold info for decimal datatype or any other datatype that requires additional information
 } gdf_dtype_extra_info;
 
+
+/**
+ * @struct gdf_scalar
+ * @brief  literal or variable
+ *
+ * The struct is used as a literal or a variable in the libgdf library.
+ *
+ * @var data     A union that represents the value.
+ * @var dtype    An enum that represents the type of the value.
+ * @var is_valid A boolean that represents whether the scalar is null.
+ */
+typedef struct {
+    gdf_data  data;
```

Comment:
I think instead of a union, we should be just using a `void*` to a single element to be consistent with `gdf_column`.

Reply:
good point, I think we should also remove the `is_valid` member from a `gdf_scalar`. Because why would you pass a singular value if it's null? One reason could be that blazing thought this could be a way to return values of reductions.

Reply:
Yeah, I also had trouble thinking of why you'd ever want a null scalar.